src/esp/assets/ResourceManager.cpp

```diff
@@ -58,6 +63,231 @@ bool ResourceManager::loadScene(const AssetInfo& info,
+          loadGeneralMeshData(info, parent, drawables, true, *physNode);
+      bool objectSuccess;
+      // Load convex hull
+      std::string fname;
+      fname.assign(info.filepath);
+      std::string c_fname =
+          //fname.replace(fname.end() - 4, fname.end(), "VHACD.glb");
+          fname.replace(fname.end() - 4, fname.end(), "_convex.glb");
+      AssetInfo c_info = AssetInfo::fromPath(c_fname);
+
+      // Load collision mesh, need to shift to origin
+      bool shift_Origin = true;
+      bool c_MeshSuccess = loadGeneralMeshData(c_info, parent, nullptr, shift_Origin);
+
+      // const std::string& filename = info.filepath;
+      MeshMetaData& metaData = resourceDict_.at(c_fname);
+      auto indexPair = metaData.meshIndex;
+      int start = indexPair.first;
+      int end = indexPair.second;
+      LOG(INFO) << "Accessing object mesh start " << start << " end " << end;
+
+      //std::vector<std::shared_ptr<BaseMesh>> objectMesh(&meshes_[start], &meshes[end]);
+      std::vector<Magnum::Trade::MeshData3D*> meshGroup;
+      for (int mesh_i = start; mesh_i < end; mesh_i++) {
+        GltfMeshData* gltfMeshData = dynamic_cast<GltfMeshData*>(meshes_[mesh_i].get());
+        auto& meshData = gltfMeshData->getMeshData();
+        meshGroup.push_back(&meshData);
+      }
+      
+      transformAxis(info, meshGroup);
+      objectSuccess = _physicsManager.initObject(info, metaData, meshGroup, *physNode);
+
+      (*physNode)->MagnumObject::setTransformation(parent->transformation());
+      (*physNode)->syncPose();
+      //LOG(INFO) << "Parent trans " << Eigen::Map<mat4f>(parent->transformation().data());
+      // PhysicsManager::initObject(*importer, info, mMetaData);
+      // for (int index = 0; index < meshes_.size(); index++) {
+      //
+      return c_MeshSuccess && objectSuccess;
+
+    } else {
+      // Objects in other formats not supported yet
+      return false;
+    }
+  } else {
+    bool meshSuccess = loadGeneralMeshData(info, parent, drawables, true);
+    return meshSuccess;
+  }
+}
+
+void ResourceManager::transformAxis(
+    const AssetInfo& info,
+    std::vector<Magnum::Trade::MeshData3D*> meshGroup) {
+
+  quatf quatf = quatf::FromTwoVectors(info.frame.front(), geo::ESP_FRONT);
+  Magnum::Quaternion quat = Magnum::Quaternion(quatf);
+  for (int iOMesh = 0; iOMesh < meshGroup.size(); iOMesh++) {
+    Magnum::Trade::MeshData3D* meshData = meshGroup[iOMesh];
+    Magnum::Matrix4 transform =
+        Magnum::Matrix4::rotation(quat.angle(), quat.axis().normalized());
+    Magnum::MeshTools::transformPointsInPlace(transform,
+                                              meshData->positions(0));
+  }
+}
+
+void ResourceManager::shiftMeshDataToOrigin(GltfMeshData* meshDataGL) {
+  Magnum::Trade::MeshData3D& meshData = *(meshDataGL->getMeshData());
+  float minX = 999999.9f;
+  float maxX = -999999.9f;
+  float minY = 999999.9f;
+  float maxY = -999999.9f;
+  float minZ = 999999.9f;
+  float maxZ = -999999.9f;
```

Comment:
This can be done using Magnum's eigen intergation with the following pattern:
```
Eigen::Array3f min = <whatever magnum's method is>(meshData.positions(0)[0]).array()
...
for ()
    min = min.cwiseMin(<whatever magnum's method is>(meshData.positions(0)[vi]).array())
```

Reply:
Or just use Magnum and do it in an oneliner :P 

```cpp
std::pair<Vector3, Vector3> bounds = Math::minmax<Vector3>(meshData.positions(0));
```

Note you may need to include `<Magnum/Math/FunctionsBatch.h>` and `<Corrade/Containers/ArrayViewStl.h>` for this to work.

Reply:
Fancy :)